packages/teuchos/parameterlist/test/yaml/YamlParameterList.cpp

```diff
@@ -104,14 +104,14 @@ namespace TeuchosTests
         );
   }
 
-  TEUCHOS_UNIT_TEST(YAML, PR1805)
+  TEUCHOS_UNIT_TEST(YAML, PR1805) // "\t" has been removed since yaml does not support tabs
   {
     RCP<ParameterList> params = Teuchos::getParametersFromYamlString(
         "My Awesome Problem:\n"
-        "\tMesh:\n"
-        "\t\tInline:\n"
-        "\t\t\tType: Quad\n"
-        "\t\t\tElements: [     10,     10 ]\n"
+        "  Mesh:\n"
+        "    Inline:\n"
+        "      Type: Quad\n"
+        "      Elements: [     10,     10 ]\n"
```

Comment:
YAML does not accept tabs (only spaces), as documented [here](http://yaml.org/faq.html). Since yaml-cpp checks for proper yaml formatting before parsing, I had to change this unit test to use spaces instead of `\t`.

Reply:
According the the official standard, I believe it does support tabs:

https://yaml.org/spec/1.2.2/

See section 5.5 on whitespace characters. The link specified above looks like outdated material (the copyright is 2006). What happens with yaml cpp?

Reply:
yaml-cpp throws an `illegal map value` error when it tries to parse `\t`.

Example:
```cpp
  TEUCHOS_UNIT_TEST(YAML, PR1805)
  {
    RCP<ParameterList> params = Teuchos::getParametersFromYamlString(
        "My Awesome Problem:\n"
        "\tMesh:\n"
        "    Inline:\n"
        "      Type: Quad\n"
        "      Elements: [     10,     10 ]\n"
        "...\n"
        );
  }
```
throws
```
yaml-cpp: error at line 2, column 6: illegal map value
```

Reply:
From https://yaml.org/spec/1.2.2/ we have BOTH
A. 
"5.5. White Space Characters
YAML recognizes two white space characters: space and tab."
and B.
 "6.1. Indentation Spaces
In YAML , structure is determined by indentation. In general, indentation is defined as a zero or more characters at the start of a line.
To maintain portability, [tab](https://yaml.org/spec/1.2.2/#white-space-characters) characters must not be used in indentation, since different systems treat [tabs](https://yaml.org/spec/1.2.2/#white-space-characters) differently."

Section 6.1 makes it seem like the example here (with tabs being used as indentation of the nodes) is not correct yaml. Section 5.5 indeed says that tabs are "valid yaml" and they can be used quoted in scalars or as in section 6.2(Separation Spaces) and 6.3(Line Prefixes), but the tab use case in this test here is for indentation.

Reply:
Did a little more research on this. In sections 6.1 and 6.2 we have these comments:

To maintain portability, [tab](https://yaml.org/spec/1.2.2/#white-space-characters) characters must not be used in indentation, since different systems treat [tabs](https://yaml.org/spec/1.2.2/#white-space-characters) differently. 

The amount of indentation is a [presentation detail](https://yaml.org/spec/1.2.2/#presenting-the-serialization-tree) and must not be used to convey [content](https://yaml.org/spec/1.2.2/#nodes) information.

Outside [indentation](https://yaml.org/spec/1.2.2/#indentation-spaces) and [scalar content](https://yaml.org/spec/1.2.2/#scalar), YAML uses [white space](https://yaml.org/spec/1.2.2/#white-space-characters) characters for separation between tokens within a line. Note that such [white space](https://yaml.org/spec/1.2.2/#white-space-characters) may safely include [tab](https://yaml.org/spec/1.2.2/#white-space-characters) characters.

Example 6.1 shows where tabs can be safely used. I think the tabs before the "Inline" are safe to use, not sure if its worth fixing. I'm fine with these changes as is.